python/API_Examples_Python.py

```diff
@@ -0,0 +1,200 @@
+mainnet_url = "https://dex.top"
+user_information = {"token": "",
+                    "traderAddr": "",
+                    "email": "",
+                    "password": ""}
+
+def get_market_info():
+    """
+    Get relevant market information such as market contract address and token codes for signature.
+    """
+    resp = requests.get(mainnet_url + "/v1/market")
+    if resp.status_code == requests.codes.ok:
+        pprint.pprint(resp.json())
+    else:
+        print("ERROR: Get Market Information request did not work. ")
+
+def get_pairs_by_cash():
+    """
+    Get the real-time trading information of all available trading pairs of the specified cash token (e.g. "ETH")
+    """
+    resp = requests.get(mainnet_url + "/v1/pairlist/ETH")
+    if resp.status_code == requests.codes.ok:
+        pprint.pprint(resp.json())
+    else:
+        print("ERROR: Get Pairs by Cash request did not work. ")
+
+def get_pair_info():
+    """
+    Get the real-time information of a trading pair.
+    """
+    resp = requests.get(mainnet_url + "/v1/pairinfo/ETH_CHSB")
+    if resp.status_code == requests.codes.ok:
+        pprint.pprint(resp.json())
+    else:
+        print("ERROR: Get Pair Information did not work. ")
+
+def get_trade_history():
+    """
+    Get recent trades by pairId sort by trade time.
+    """
+    resp = requests.get(mainnet_url + "/v1/tradehistory/ETH_CHSB/9")
+    if resp.status_code == requests.codes.ok:
+        pprint.pprint(resp.json())
+    else:
+        print("ERROR: Get Trade History request did not work. ")
+
+def get_pair_Depth():
+    """""
+    # Get the depth data of a trading pair.
+    """
+    resp = requests.get(mainnet_url + "/v1/depth/ETH_YEE/10")
+    if resp.status_code == requests.codes.ok:
+        pprint.pprint(resp.json())
+    else:
+        print("ERROR: Get Pair Depth request did not work. ")
+
+def user_login():
+    """
+    API used to login to exchange - Returns a token
+    """
+    pay_load = {"email": user_information['email'], "password": user_information['password']}
+    pay_load = json.dumps(pay_load)
+    r = requests.post(mainnet_url + "/v1/authenticate", pay_load)
+    if r.status_code == requests.codes.ok:
+        resp = r.json()
+        user_information["token"] = resp['token']
+        print("Login Successful")
+    else:
+        print("ERROR: User login post did not work. ")
+
+def get_balance():
+    """
+    Get the balances of all tokens of a trader.
+    """
+    headers = {"Authorization": user_information["token"]}
+    resp = requests.get(mainnet_url + "/v1/balances/" + user_information["traderAddr"], headers=headers)
+    if resp.status_code == requests.codes.ok:
+        pprint.pprint(resp.json())
+    else:
+        print("ERROR: Get Balance request did not work. ")
+
+def place_order():
+    """
+    Place a new order
+    """
+    headers = {"Authorization": user_information["token"]}
+    signature = example_sign_order.play()
```

Comment:
You should extra a reusable order signing function `SignOrder(priv_key, order_data)`, which returns the signature. Do not import example_sign_order directly. Actually, when this full example is ready, we obviously don't need the signing example anymore and will deleted it.